<launch>
  <!-- 定义 mode 参数，默认值为 0,读取主从臂的消息，并转发到ROS -->
  <arg name="mode" default="0" />
  <arg name="auto_enable" default="true" />
  <!-- 启动左侧机械臂节点 -->
    <node name="$(anon piper_left)" pkg="piper" type="piper_start_ms_node.py" output="screen">
      <param name="can_port" value="can_left" />
      <param name="mode" value="$(arg mode)" />
      <param name="auto_enable" value="$(arg auto_enable)" />
      <remap from="/puppet/arm_status" to="/puppet/arm_status_left" />
      <remap from="/puppet/joint_states" to="/puppet/joint_left" />
      <remap from="/master/joint_states" to="/master/joint_left" />
      <remap from="/puppet/end_pose" to="/puppet/end_pose_left" />
      <remap from="/pos_cmd" to="/puppet/pos_cmd_left" />
      <remap from="/puppet/end_pose_euler" to="/puppet/end_pose_euler_left" />
    </node>

  <!-- 启动右侧机械臂节点 -->
    <node name="$(anon piper_right)" pkg="piper" type="piper_start_ms_node.py" output="screen">
      <param name="can_port" value="can_right" />
      <param name="mode" value="$(arg mode)" />
      <param name="auto_enable" value="$(arg auto_enable)" />
      <remap from="/puppet/arm_status" to="/puppet/arm_status_right" />
      <remap from="/puppet/joint_states" to="/puppet/joint_right" />
      <remap from="/master/joint_states" to="/master/joint_right" />
      <remap from="/puppet/end_pose" to="/puppet/end_pose_right" />
      <remap from="/pos_cmd" to="/puppet/pos_cmd_right" />
      <remap from="/puppet/end_pose_euler" to="/puppet/end_pose_euler_right" />
    </node>
</launch>

